cmd/talosctl/cmd/mgmt/cluster/create/common_test.go

```diff
@@ -0,0 +1,454 @@
+		input := getTestOps()
+		input.SkipK8sNodeReadinessCheck = true
+		input.SkipKubeconfig = true
+
+		_, err := _getBase(input, testProvisioner{}, getTalosTestVersion, returnNoAdditionalOpts)
+		assert.NoError(t, err)
+	})
+
+	addFieldTest("", n(&tf, &tf.NetworkIPv6), func(t *testing.T) {
+		input := getTestOps()
+		input.NetworkIPv6 = true
+
+		result, err := _getBase(input, testProvisioner{}, getTalosTestVersion, returnNoAdditionalOpts)
+		assert.NoError(t, err)
+		nodes := append(result.base.clusterRequest.Controlplanes, result.base.clusterRequest.Workers...)
+
+		for _, n := range nodes {
+			assert.Equal(t, 2, len(n.IPs))
+			assert.Equal(t, true, n.IPs[1].Is6())
+		}
+		assert.Equal(t, "10.5.0.2", nodes[0].IPs[0].String())
+		assert.Equal(t, "fd74:616c:a05::2", nodes[0].IPs[1].String())
+		assert.Equal(t, "10.5.0.5", nodes[3].IPs[0].String())
+		assert.Equal(t, "fd74:616c:a05::5", nodes[3].IPs[1].String())
+	})
+	addFieldTest("", n(&tf, &tf.InputDir), func(t *testing.T) {
+		dir := t.TempDir()
+		input := getTestOps()
+		input.RootOps.StateDir = dir
+		input.RegistryMirrors = []string{"test.test=https://test.mirror"}
+		result, err := _getBase(input, testProvisioner{}, getTalosTestVersion, returnNoAdditionalOpts)
+		assert.NoError(t, err)
+		result.configBundle.Write(dir, encoder.CommentsDisabled, machine.TypeControlPlane, machine.TypeWorker)
+		assert.NoError(t, err)
+
+		input = getTestOps()
+		input.InputDir = dir
+		fmt.Println(dir)
+		result, err = _getBase(input, testProvisioner{}, getTalosTestVersion, returnNoAdditionalOpts)
+		assert.NoError(t, err)
+		assert.EqualValues(t, "https://test.mirror", result.base.clusterRequest.Workers[0].Config.RawV1Alpha1().MachineConfig.Registries().Mirrors()["test.test"].Endpoints()[0])
+		assert.EqualValues(t, "https://test.mirror", result.base.clusterRequest.Controlplanes[0].Config.RawV1Alpha1().MachineConfig.Registries().Mirrors()["test.test"].Endpoints()[0])
+	})
+}
+
+func getGenOpts(t *testing.T, input CommonOps) generate.Options {
+	baseResult, err := _getBase(input, testProvisioner{}, getTalosTestVersion, returnNoAdditionalOpts)
+	assert.NoError(t, err)
+	result, err := generate.NewInput(input.RootOps.ClusterName, "cluster.endpoint", "k8sv1", baseResult.genOptions...)
+	assert.NoError(t, err)
+	return result.Options
+}
+
+func addFieldTest(testName string, fieldNames []string, test func(t *testing.T)) {
+	testedFields = append(testedFields, fieldNames...)
+	name := fmt.Sprintf("Test%sField", fieldNames[0])
+	if testName != "" {
+		name = testName
+	}
+	if len(fieldNames) > 1 && testName == "" {
+		panic("no test name specified with multiple fields tested")
+	}
+
+	fieldTests = append(fieldTests, fieldTest{fieldNames, test, name})
+}
+
+var fieldTests []fieldTest
+
+type fieldTest = struct {
+	fieldNames []string
+	test       func(t *testing.T)
+	name       string
+}
+
+var testedFields = []string{}
+
+func TestAllCommonOptionFields(t *testing.T) {
```

Comment:
Test that there are tests for all the common fields. Please let me know if this is inappropriate :) or if there's a less hacky way to achieve the same result.